### ISSUE TITLE
[BUG] [Linter] H025 false positive in 1.43.0: closing tag of every multi-line <script>/<style> reports as an orphan

### ISSUE THREAD

**Issue by safetymonkey** [djlint/djLint#2302]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OS: Ubuntu 24.04.4 LTS
- Python Version: 3.14.3
- djLint Version: 1.43.0
- template language: jinja (also reproduces with the default profile)

## Issue

In 1.43.0, `H025` ("Tag seems to be an orphan") fires on the closing tag of **any** `<script>` or `<style>` element whose opening and closing tags are not on the same source line. Since many real-world script block spans multiple lines, this makes H025 noisy at best and arguably unusable at worst. It reported 25 errors on my project, all on `</script>`, and every one of them false.

1.42.1, 1.42.2 and 1.42.3 are all clean on the same input.

### Affected / unaffected

| Input | 1.42.3 | 1.43.0 |
|---|---|---|
| `<script>` … `</script>` on one line | ok | ok |
| `<script src="a.js"></script>` | ok | ok |
| Multi-line `<script>` (inline JS) | ok | **H025** |
| Multi-line `<script>` open tag (`src` + `integrity` + `crossorigin`) | ok | **H025** |
| `<script type="application/ld+json">` block | ok | **H025** |
| Multi-line `<style>` | ok | **H025** |
| Multi-line `<pre>` / `<textarea>` | ok | ok |

Nesting depth and attributes make no difference; the only thing that matters is
whether the opening and closing tags share a line.

## How To Reproduce

No config file and no flags are required.

1. Create `repro.html`:

```html
<div>
  <script>
  var x = 1;
  </script>
</div>
```

2. Run djLint against it:

```console
$ uv tool run djlint==1.42.3 --lint repro.html
Linted 1 file, found 0 errors.

$ uv tool run djlint==1.43.0 --lint repro.html
repro.html
───────────────────────────────────────────────────────────────────────────────
H025 4:2 Tag seems to be an orphan. </script>

Linted 1 file, found 1 error.
```

**Expected:** 0 errors, as in 1.42.3 — the `<script>` element is well-formed and its
tags are balanced.

**Actual:** 1.43.0 reports H025 on the `</script>`.

Swapping the `<script>` for a multi-line `<style>` produces the same result on
`</style>`. Collapsing the element onto a single line
(`<script>var x = 1;</script>`) makes the error disappear.

## Contents of djlint.toml/.djlintrc/pyproject.toml [tool.djlint]

None — the reproduction above uses no config file at all and no `--profile` flag, so it
runs entirely on djLint's defaults.

For completeness, the project where we hit this uses:

```toml
[tool.djlint]
indent = 2
profile = "jinja"
max_line_length = 120
format_css = false
format_js = false
max_attribute_length = 70
format_attribute_template_tags = false
ignore = "H006,H030,H031,H029,J018,H021,T028"
extend_exclude = "_privacy_static.html|_terms_static.html|_faq_static.html"
```

The false positive is not sensitive to any of these settings; it reproduces with the
defaults.

## Root cause (according to Claude)
Please feel free to ignore if you don't want the AI's opinion.

`rules/H025.py` is byte-identical between 1.42.3 and 1.43.0. The behavior change comes
from commit 97d8737, *"fix(linter): don't treat a tag touching an ignored block as
inside it"* (fix for #2297, a T038 false positive on inline comments), which changed
`overlaps_ignored_block()` in `src/djlint/helpers.py` from inclusive to half-open span
comparisons:

```diff
-        if (ignored_match_start <= match_start <= ignored_match_end) or (
-            ignored_match_start <= match_end <= ignored_match_end
+        # spans are half open, so a match that only touches an ignored block
+        # (e.g. `{% if x %}{# comment #}`) starts and ends outside of it.
+        if (ignored_match_start <= match_start < ignored_match_end) or (
+            ignored_match_start < match_end <= ignored_match_end
         ):
```

The interaction with the script/style ignore pattern (`settings.py`, ~line 804) is what
breaks it. That pattern terminates with a **lookahead**:

```
| <(script|style).*?(?=(\</(?:\3)>))
```

so the ignored span ends exactly at the `<` of the closing tag. That gives
`match_start == ignored_match_end` for `</script>`. Under the old inclusive test the
closer counted as inside the ignored block and was skipped; under the new half-open
test it is not skipped — while its opening `<script>` still is, because that sits at
`ignored_match_start`. The pair becomes asymmetric, so H025 sees a closer with no
matching opener and reports an orphan.

`<pre>` and `<textarea>` are unaffected because their pattern on the preceding line
(`<(pre|textarea).*?</(\1)>`) includes the closing tag inside the span, so the
half-open comparison still matches.

The 1.43.0 release notes do flag H025 as touched by this change, under **Fix**:

> A tag that merely touches an ignored block is no longer treated as being inside it.
> A tag ending exactly where an ignored block starts - such as `{% if x %}{# comment #}`
> - was skipped by the linter, most visibly making `T038` report `End tag has no
> matching block tag` for a balanced `{% if %}`. Affects every rule that skips ignored
> blocks (`H025`, `H037`, `H041`, `H042`, `T002`, `T003`, `T027`, `T038`, `T039`) and
> every kind of ignored block.

So the effect on H025 was anticipated in the abstract. What appears not to have been
anticipated is the interaction with the script/style **lookahead** specifically: the
change fixes the `{% if x %}{# comment #}` case at the *start* boundary, but at the
*end* boundary it silently breaks the far more common multi-line `<script>`, because
that pattern's span stops short of the closing tag where `pre`/`textarea`'s does not.

Worth noting the same code path is shared with `H037`, `H041`, `H042`, `T002`, `T003`,
`T027`, `T038` and `T039`. Only H025 surfaced on our corpus of 109 templates, but the
others may deserve a look for the same end-boundary case.

### Suggested fix

Either make the script/style ignore pattern consume its closing tag (matching how
`pre`/`textarea` are handled), or special-case the boundary so a tag starting exactly
at `ignored_match_end` is still treated as part of the block.

A regression test with a multi-line `<script>` would cover this. Commit 97d8737 added
tests only to `tests/test_linter/test_t038.py`, all for the `{# comment #}` start-boundary
case in #2297 — no test exercises H025, and no added test input contains a `<script>` or
`<style>` element at all, which is why this slipped through.
